freedomofpress/securedrop-workstation
# Tweak resize behavior to avoid text cutoff

## Status

Ready for review

## Description of Changes

Fixes #597. Enforces a maximum and minimum size while allowing text to take up as much space as needed, even if font size changes.

- Assign a grid layout to the QDialog to ensure it reflows with resize
- Match QDialog class to generated UI code in multiple inheritance
- Set size policy for long text to "minimum" to ensure it always allocates as much space as needed
- Set a maximum to allow for some adjustment by the user

## Testing
Default font size:
1. Apply the changes in this PR to the launcher versions in `/opt/securedrop/launcher` and `/srv/salt/launcher` (if only the `/opt` copy is overwritten, the updater itself will replace it on the next run).
2. Run  `/opt/securedrop/launcher/sdw-launcher.py --skip-delta 0`. This forces an updater run.
3. While the updater runs, attempt to resize the window. Note that you can do so up to a maximum and minimum size. 
- [ ] Observe that no text has been cut off.

Increased font size:
1. Under System->Appearance in dom0, increase the "Default Font" to a very high value, e.g., `17`.
2. Re-run the updater as above.
- [ ] Observe that no text has been cut off.

## Checklist

- [x] Linter (`make flake8`) passes in the development environment
